packages/orm/src/EntityManager.ts

```diff
@@ -398,6 +401,90 @@ export class EntityManager<C = {}> {
     return createOrUpdatePartial(this, type, opts);
   }
 
+  /**
+   * Utility to clone an entity and its nested relations, as determined by a populate hint
+   *
+   * @param entity - Any entity
+   * @param hint - A populate hint
+   *
+   * @example
+   * // This will duplicate the author
+   * const duplicatedAuthor = await em.clone(author)
+   *
+   * @example
+   * // This will duplicate the author and all their related book entities
+   * const duplicatedAuthorAndBooks = await em.clone(author, "books")
+   *
+   * @exmaple
+   * // This will duplicate the author, all their books, and the images for those books
+   * const duplicatedAuthorAndBooksAndImages = await em.clone(author, {books: "image"})
+   */
+  public async clone<T extends Entity, H extends LoadHint<T>>(entity: T, hint?: H): Promise<Loaded<T, H>> {
+    const meta = getMetadata(entity);
+    const { id, ...data } = entity.__orm.data;
+    const clone = this.create(meta.cstr, {} as OptsOf<T>);
```

Comment:
Fwiw @zgavin I assume doing this didn't work:

```typescript
const { id, ...data } = entity.__orm.data;
const clone = this.create(meta.cstr, data);
```

I.e. passing `data` directly to `create`? Would be nice to have an in-source comment if there is a particularly tricky reason about why that doesn't work, but not a big deal.